userspace/libsinsp/user.cpp

```diff
@@ -256,9 +256,32 @@ scap_userinfo *sinsp_usergroup_manager::userinfo_map_insert(
 	auto &usr = map[uid];
 	usr.uid = uid;
 	usr.gid = gid;
-	strlcpy(usr.name, name, MAX_CREDENTIALS_STR_LEN);
-	strlcpy(usr.homedir, home, SCAP_MAX_PATH_SIZE);
-	strlcpy(usr.shell, shell, SCAP_MAX_PATH_SIZE);
+	// In case the node is configured to use NIS,
+	// some struct passwd* fields may be set to NULL.
+	if(!name)
+	{
+		strlcpy(usr.name, "<NA>", MAX_CREDENTIALS_STR_LEN);
+	}
+	else
+	{
+		strlcpy(usr.name, name, MAX_CREDENTIALS_STR_LEN);
+	}
+	if(!home)
+	{
+		strlcpy(usr.homedir, "<NA>", MAX_CREDENTIALS_STR_LEN);
+	}
+	else
+	{
+		strlcpy(usr.homedir, home, SCAP_MAX_PATH_SIZE);
+	}
+	if(!shell)
+	{
+		strlcpy(usr.shell, "<NA>", MAX_CREDENTIALS_STR_LEN);
+	}
+	else
+	{
+		strlcpy(usr.shell, shell, SCAP_MAX_PATH_SIZE);
+	}
```

Comment:
I'd like it better with ternary :)
```suggestion
	strlcpy(usr.name, (name != nullptr) ? name : "<NA>", MAX_CREDENTIALS_STR_LEN);
	strlcpy(usr.homedir, (home != nullptr) ? home : "<NA>", SCAP_MAX_PATH_SIZE);
	strlcpy(usr.shell, (shell != nullptr) ? shell : "<NA>", SCAP_MAX_PATH_SIZE);
```
Also mind the mixed use of `MAX_CREDENTIALS_STR_LEN` and `SCAP_MAX_PATH_SIZE`

Reply:
Much nicer, thank you! And thanks also @wigol for the fix!

Reply:
Waiting for it upstream :)

Reply:
@FedeDP the upstream change is in: https://github.com/falcosecurity/libs/pull/927